storage.go

```diff
@@ -20,6 +20,8 @@ type Storage interface {
 	UploadData(data []byte, storagePath, contentType string) (location string, size int64, err error)
 	UploadFile(filepath, storagePath, contentType string) (location string, size int64, err error)
 
+	ListObjects(prefix string) ([]string, error)
```

Comment:
Not a public API so something we can change eventually, but in my experience, returning a channel where entries are posted as each page comes back ends up being more usable for large lists.